src/components/frame/v5/pages/BalancePage/partials/BalanceTable/hooks.tsx

```diff
@@ -171,3 +176,137 @@ export const useGetTableMenuProps = () =>
+    type: {},
+    attribute: {
+      native: false,
+      reputation: false,
+    },
+  });
+
+  const tokensData = useMemo(
+    () =>
+      colony?.tokens?.items.map((item) => {
+        const currentTokenBalance =
+          getBalanceForTokenAndDomain(
+            balances,
+            item?.token?.tokenAddress || '',
+            selectedTeam ? Number(selectedTeam.nativeId) : undefined,
+          ) || 0;
+        const decimals = getTokenDecimalsWithFallback(item?.token.decimals);
+        const convertedValue = convertToDecimal(
+          currentTokenBalance,
+          decimals || 0,
+        );
+
+        const formattedValue = getFormattedNumeralValue(
+          convertedValue,
+          currentTokenBalance,
+        );
+
+        return {
+          ...item,
+          balance: typeof formattedValue === 'string' ? formattedValue : '',
+        };
+      }),
+    [colony, balances, selectedTeam],
+  );
+
+  const sortedTokens = useMemo(
+    () =>
+      tokensData?.sort((a, b) => {
+        if (!a.balance || !b.balance) return 0;
+        return parseInt(b.balance, 10) - parseInt(a.balance, 10);
+      }),
+    [tokensData],
+  );
+
+  const tokenTypeFilters = sortedTokens
+    ? sortedTokens.map(({ token }) => ({
+        name: token?.tokenAddress || '',
+        label: token && (
+          <div className="flex items-center gap-2">
+            <TokenIcon token={token} size="xxxs" />
+            {token.symbol}
+          </div>
+        ),
+      }))
+    : [];
+
+  const filters: FilterProps<BalanceTableFilters> = {
+    onChange: setValue,
+    onSearch: setSearchValue,
+    searchValue,
+    value,
+    customLabel: formatText({ id: 'allFilters' }),
+    items: [
+      {
+        name: 'type',
+        label: formatText({ id: 'balancePage.filter.tokenType' }),
+        title: formatText({ id: 'balancePage.filter.approvedTokens' }),
+        iconName: 'coin-vertical',
+        items: tokenTypeFilters,
+      },
+      {
+        name: 'attribute',
+        label: formatText({ id: 'balancePage.filter.attributes' }),
+        title: formatText({ id: 'balancePage.filter.attributeTypes' }),
+        iconName: 'folder-simple-lock',
+        items: [
+          {
+            name: 'native',
+            label: formatText({ id: 'token.type.native' }),
+          },
+          {
+            name: 'reputation',
+            label: formatText({ id: 'token.type.reputation' }),
+          },
+        ],
+      },
+    ],
+  };
+
+  const tokens = sortedTokens?.map((token) => ({
+    ...token,
+    isApproved: sortedTokens?.some(
+      ({ token: colonyToken }) =>
+        colonyToken?.tokenAddress === token?.token?.tokenAddress,
+    ),
+    isNative: token.token?.tokenAddress === nativeToken?.tokenAddress,
+  }));
+
+  const visibleTokens = tokens?.filter((visibleToken) => {
+    const { type = {}, attribute } = value;
+
+    if (Object.values(type).some((tokenTypeFilter) => tokenTypeFilter)) {
```

Comment:
`if (Object.values(type).length > 0)` ?

Reply:
and imo we need better refined types here, I am not 100% sure what happens here

Reply:
we can't change it in line 287 to this condition` if (Object.values(type).length > 0) ?`
Then this condition is true then we will return filtered data, otherwise all data will be returned. When we change to the condition which you suggested then no data will be displayed as default

Reply:
ah OK my bad, I misunderstood this codesnip then, we want to check if there are any filters that have a value of true